src/Cardano/Wallet/Binary/Helpers.hs

```diff
@@ -0,0 +1,66 @@
+{-# LANGUAGE RankNTypes #-}
+
+-- | Some extra helpers on top of the CBOR library to help writing decoders.
+
+module Cardano.Wallet.Binary.Helpers
+    ( inspectNextToken
+    , decodeList
+    , decodeListIndef
+    , unsafeDeserialiseFromBytes
+    ) where
+
+import           Prelude
+import qualified Codec.CBOR.Decoding              as CBOR
+import qualified Codec.CBOR.Read                  as CBOR
+import qualified Data.ByteString.Lazy             as BL
+import           Debug.Trace                      (traceShow)
+
+-- | Inspect the next token that has to be decoded and print it to the console
+-- as a trace. Useful for debugging Decoders.
+-- Example:
+--
+--     myDecoder :: CBOR.Decoder s MyType
+--     myDecoder = do
+--         a <- CBOR.decodeWord64
+--         inspectNextToken
+--         [...]
+--
+inspectNextToken :: CBOR.Decoder s ()
+inspectNextToken =
+  CBOR.peekTokenType >>= flip traceShow (return ())
```

Comment:
`HLint` is probably going to complain about the `traceShow`, we may require an annotation up there ^^^